### PR TITLE
Fixes macOS build

### PR DESCRIPTION
**Pull request by honghaoz**
Following upon https://github.com/devicekit/DeviceKit/pull/342, to address: https://github.com/devicekit/DeviceKit/issues/341

Changes:
- Instead of excluding the whole library for macOS, this PR tries to conditionally compile the code for macOS.
- ~~Bumps up the SPM tool version to 5.8, to support the privacy manifest file~~
- Fixes unit tests on macOS

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.0
+// swift-tools-version:5.8
```
Comment (Lutzifer) [inline Package.swift:1]: just wanted to propose this change, thank you!

Comment (Zandor300) [reply Package.swift:1]: Thanks, but going in favor of https://github.com/devicekit/DeviceKit/pull/381 for this specific change.

**Review by Zandor300 — CHANGES_REQUESTED** [review]
Please rebase your PR on master, I don't have permission to push to your PR branch.

Please use swift-tools-version 5.3 from master.

**Comment by honghaoz** [comment]
@Zandor300 PR updated. Thanks 🙏

**Review by Zandor300 — APPROVED** [review]

**Pull request closed without merging**